mmv1/templates/terraform/decoders/vertex_ai_models.go.tmpl

```diff
@@ -0,0 +1,10 @@
+// Handle the name field not being set in the API response after Create (is set after Read)
```

Comment:
Note: This is necessary in the decoder because the generated code for the Create method forces the resource's id to be set using a name field:


```go
	if err := d.Set("name", flattenVertexAIModelName(opRes["name"], d, config)); err != nil {
		return err
	}

	// This may have caused the ID to update - update it if so.
	id, err = tpgresource.ReplaceVars(d, config, "{{name}}")
	if err != nil {
		return fmt.Errorf("Error constructing id: %s", err)
	}
	d.SetId(id)
```

We could set `api_name: 'model'` in the Model.yaml file for the name field, but that would instead mean that the decoder would need to use the name field to fake the presence of the model field on Read; we're stuck between how `name` is not present after Create but is present after Read, and `model` is present after Create but not present after Read.

Reply:
This could be addressed by setting `id_format: 'projects/{{project}}/locations/{{region}}/models/{{model_id}}'` in the YAML, but then the model_id is not available at the right time in the Create method.

[We could make the Create method entirely custom code](https://googlecloudplatform.github.io/magic-modules/develop/custom-code/#replace-entire-crud-methods) to overcome this. What do you think @BBBmau?

Reply:
I don't think it's worth committing to writing the custom_create just to resolve this only because of the overhead that it comes from it especially when dealing with what is to come when needing to support UPLOAD and all it's other fields. 

I think it's a fine trade-off of needing to address the missing name field on Create, a well-written comment should be more than enough so that any future reviewer can quickly understand why this is implemented the way it is.